File: prove/grpc/javapy/java/src/main/java/client/MicroClient.java
package client;

import java.util.concurrent.TimeUnit;

import io.grpc.StatusRuntimeException;
import io.grpc.Channel;
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;

import client.proto.*;

import client.proto.GreeterGrpc.GreeterBlockingStub;
import client.proto.GreeterGrpc.GreeterStub;

public class MicroClient {

    private final GreeterBlockingStub blockingStub;
    private final GreeterStub asyncStub;

    public MicroClient(Channel channel) {
        blockingStub = GreeterGrpc.newBlockingStub(channel);
        asyncStub = GreeterGrpc.newStub(channel);
    }

    public void getResponse(String id, String name) {
        Request input = Request.newBuilder().setId(id).setName(name).build();

        Response output;
        try {
            output = blockingStub.getServiceName(input);
            System.out.println("Response: " + output);

        } catch (StatusRuntimeException e) {
            System.out.println("Something went wrong during gRPC communication.");
        }

    }

    public static void main(String[] args) throws InterruptedException {
        //open gRPC channel
        String target = "localhost:50051";
        ManagedChannel channel = ManagedChannelBuilder.forTarget(target).usePlaintext().build();

        try {
            //create client stub
            MicroClient client = new MicroClient(channel);

            //get response
            client.getResponse("1", "What's your name?");          

        } finally {
           channel.shutdownNow().awaitTermination(5, TimeUnit.SECONDS); 
        }

    }

}

File: prove/docker/pypy/client/MicroClient.py
import grpc

from proto import ClientToServer_pb2
from proto import ClientToServer_pb2_grpc

ADDR_PORT = 'localhost:50051'   #server_IP_addr:port_num

def run():
    #open gRPC channel
    channel = grpc.insecure_channel(ADDR_PORT)  #server_IP_addr:port_num
    #channel = grpc.insecure_channel('172.30.160.1:50051', options=(('grpc.enable_http_proxy', 0),))

    #create client stub
    stub = ClientToServer_pb2_grpc.GreeterStub(channel)

    #get response
    output = stub.GetServiceName(ClientToServer_pb2.Request(id = "1", name = "What's your name?"))
    print("Response: ", output)

run()
